### PR TITLE
Breaking Change: Don't use `isapprox` in `==`

### PR DESCRIPTION
**Pull request by eliascarv**

### CODE REVIEW

File: src/primitives/ray.jl [hunk 1]
```diff
@@ -49,4 +49,5 @@ direction(r::Ray) = r.v
 
 Base.in(p::Point, r::Ray) = p ∈ Line(r.p, r.p + r.v) && (p - r.p) ⋅ r.v ≥ 0
 
+# Decision: use isapprox 
 ==(r₁::Ray, r₂::Ray) = (r₁.p == r₂.p) && (r₁.p + r₁.v) ∈ r₂ && (r₂.p + r₂.v) ∈ r₁
```
Comment (eliascarv) [inline src/primitives/ray.jl]: Decision: use `isapprox`?

Comment (juliohm) [reply src/primitives/ray.jl]: Yes, it would be great to use isapprox explicitly here.

File: src/intersections.jl [hunk 2]
```diff
@@ -203,7 +203,7 @@ function hasintersect(g1::Geometry{Dim,T}, g2::Geometry{Dim,T}) where {Dim,T}
 
   # initial direction
   c1, c2 = centroid(g1), centroid(g2)
-  d = c1 == c2 ? rand(Vec{Dim,T}) : c2 - c1
+  d = c1 ≈ c2 ? rand(Vec{Dim,T}) : c2 - c1
```
Comment (juliohm) [inline src/intersections.jl:206]: I think we need to always use the verbose version with atol=atol(T) given that our atol is different than the default atol=0

File: src/points.jl [hunk 3]
```diff
@@ -136,31 +136,23 @@ at a reference (or start) point `A`.
 Tells whether or not the coordinates of points `A` and `B`
 are approximately equal.
 """
-Base.isapprox(A::Point, B::Point; kwargs...) =
-  isapprox(A.coords, B.coords; kwargs...)
+Base.isapprox(A::Point, B::Point; atol=atol(coordtype(A)), kwargs...) =
```
Comment (juliohm) [inline src/points.jl]:
```suggestion
Base.isapprox(A::Point{Dim,T}, B::Point{Dim,T}; atol=atol(T), kwargs...) where {Dim,T} =
```

File: test/mesh.jl [hunk 4]
```diff
@@ -172,7 +172,7 @@
     @test measure(grid) ≈ T(1)
     @test centroid(grid, 1) == P2(0.1, 0.05)
     @test centroid(grid[1]) == P2(0.1, 0.05)
-    @test centroid(grid, 2) == P2(0.3, 0.05)
+    @test centroid(grid, 2) ≈  P2(0.3, 0.05)
     @test centroid(grid[2]) == P2(0.3, 0.05)
```
Comment (juliohm) [inline test/mesh.jl]: Given that we are changing one line, maybe we should change the entire block for consistency.

File: test/transforms.jl [hunk 5]
```diff
@@ -57,7 +57,7 @@
     vnew, cache = TB.apply(trans, vset)
     @test all(sides(boundingbox(vnew)) .≤ T(1))
     vini = TB.revert(trans, vnew, cache)
-    @test vini == vset
+    @test all(isapprox.(vini, vset))
```
Comment (juliohm) [inline test/transforms.jl]:
```suggestion
    @test all(≈, vini, vset)
```

**Review by juliohm — CHANGES_REQUESTED** [review]

File: test/transforms.jl (hunk 5 above)
Comment (eliascarv) [reply test/transforms.jl]: The `all` function does not have this method, just:
```julia
all(itr)
all(f, itr)

File: src/points.jl [hunk 6]
```diff
@@ -136,31 +136,26 @@ at a reference (or start) point `A`.
 Tells whether or not the coordinates of points `A` and `B`
 are approximately equal.
 """
-Base.isapprox(A::Point, B::Point; kwargs...) =
-  isapprox(A.coords, B.coords; kwargs...)
+Base.isapprox(A::Point{Dim,T}, B::Point{Dim,T}; atol=atol(T), kwargs...) where {Dim,T} =
+  isapprox(A.coords, B.coords; atol, kwargs...)
```
Comment (juliohm) [inline src/points.jl]:
```suggestion
Base.isapprox(A::Point{Dim,T}, B::Point{Dim,T};
              atol=atol(T), kwargs...) where {Dim,T} =
  isapprox(A.coords, B.coords; atol, kwargs...)
```

File: src/points.jl [hunk 7]
```diff
@@ -136,31 +136,26 @@ at a reference (or start) point `A`.
...
+
+Base.isapprox(A::Point{Dim,T}, B::Point{Dim,S}; atol=atol(promote_type(T, S)), kwargs...) where {Dim,T,S} =
+  isapprox(A.coords, B.coords; atol, kwargs...)
```
Comment (juliohm) [inline src/points.jl]:
```suggestion
```

File: test/transforms.jl (hunk 5 above)
Comment (juliohm) [inline test/transforms.jl]:
```suggestion
    @test all(vini .≈ vset)
```

**Comment by kylebeggs** [comment]
relevant to #382 just linking for my future self!

File: src/points.jl [hunk 8]
```diff
@@ -136,31 +136,24 @@ at a reference (or start) point `A`.
 Tells whether or not the coordinates of points `A` and `B`
 are approximately equal.
 """
-Base.isapprox(A::Point, B::Point; kwargs...) =
-  isapprox(A.coords, B.coords; kwargs...)
+Base.isapprox(A::Point{Dim,T}, B::Point{Dim,T};
+              atol=atol(T), kwargs...) where {Dim,T} =
```
Comment (juliohm) [inline src/points.jl:140]:
```suggestion
Base.isapprox(A::Point{Dim,T}, B::Point{Dim,V};
              atol=atol(promote_type(T,V)), kwargs...) where {Dim,T,V} =
```

File: src/points.jl [hunk 9]
```diff
@@ -136,31 +136,24 @@ at a reference (or start) point `A`.
 Tells whether or not the coordinates of points `A` and `B`
 are approximately equal.
 """
-Base.isapprox(A::Point, B::Point; kwargs...) =
-  isapprox(A.coords, B.coords; kwargs...)
+Base.isapprox(A::Point{Dim,T}, B::Point{Dim,V};
+              atol=atol(promote_type(T,V)), kwargs...) where {Dim,T,V} =
```
Comment (juliohm) [inline src/points.jl]:
```suggestion
Base.isapprox(A::Point{Dim,T}, B::Point{Dim,T};
              atol=atol(T), kwargs...) where {Dim,T} =
```

**Comment by juliohm** [comment]
@eliascarv can you please rebase this PR to the latest master? We have some new code that was merged that is using the == currently.

**Comment by eliascarv** [comment]
Closed in favor of #392

**Pull request closed without merging**